domain/model/bootstrap/bootstrap.go

```diff
@@ -72,19 +78,79 @@ func CreateModel(
 	}
 }
 
-// CreateReadOnlyModel creates a new model within the model database with all of
-// its associated metadata. The data will be read-only and cannot be modified
-// once created.
-func CreateReadOnlyModel(
-	args model.ReadOnlyModelCreationArgs,
-) func(context.Context, database.TxnRunner) error {
-	return func(ctx context.Context, db database.TxnRunner) error {
-		if err := args.Validate(); err != nil {
-			return fmt.Errorf("model creation args: %w", err)
+// CreateReadOnlyModelInfo
+func CreateReadOnlyModelInfo(
```

Comment:
@tlm @wallyworld @SimonRichardson I was never particularly happy with what we did for `BootstrapConcern` and hoped that we'd rethink it, but time has got away and we're now trying to do crazy stuff like this with it.

What about we go back to the simple bootstrap options pattern.
- Initialisation just creates both databases and a runner for each.
- Each bootstrap option becomes:
```go 
func(context.Context, coredatabase.TxnRunner, coredatabase.TxnRunner) error
```
- So the bootstrap logic passes both runners, and the options use one or both at need.

I believe this will remove the dichotomy and simplify the logic.

Reply:
See: https://github.com/juju/juju/pull/17068